src/net_processing.cpp

```diff
@@ -3505,7 +3531,8 @@ void PeerManagerImpl::ProcessMessage(CNode& pfrom, const std::string& msg_type,
         {
         LOCK(cs_main);
 
-        if (!m_chainman.m_blockman.LookupBlockIndex(cmpctblock.header.hashPrevBlock)) {
+        const CBlockIndex* prev_header_index = m_chainman.m_blockman.LookupBlockIndex(cmpctblock.header.hashPrevBlock);
+        if (!prev_header_index || !BlockRequestAllowed(prev_header_index, /* allow_potentially_invalid_headers= */ true)) {
```

Comment:
Here and above you might add a `LogPrint` notifying a non-allowed header has been denied of request, like the others callsites of `BlockRequestAllowed` ("ignoring request from peer=%i for old blocks that isn't in the main chain"). Nice to monitor eventual fingerprinting attacks attempts or other anomalies.

Reply:
With the new approach the log message reflect whether the previous block existed or if the peer was not allowed to know about it (potential fingerprint).